socket.hpp

```diff
@@ -6,10 +6,11 @@
 
 namespace tram {
 class Socket {
-  const int m_socket_desc;
+  int m_socket_desc;
 public:
   Socket(int domain, int type, int protocol = 0);
   Socket(int socket_desc);
```

Comment:
As `Socket` can be converted to int, I believe `Socket(int)` should be private to prevent 2 `Socket`s from sharing the same socket descriptor.